pkgs/development/tools/yj/default.nix

```diff
@@ -1,20 +1,20 @@
-# This file was generated by https://github.com/kamilchm/go2nix v1.3.0
-{ stdenv, buildGoPackage, fetchgit, fetchhg, fetchbzr, fetchsvn }:
+{ stdenv, buildGoModule, fetchFromGitHub }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "yj";
-  version = "4.0.0";
-  rev = "d9a48607cc5c812e8cf4abccc8ad26f37ab51558";
+  version = "5.0.0";
+  rev = "c4c13b7641389c76ea028b48091f851f3efb6376";
 
-  goPackagePath = "github.com/sclevine/yj";
-
-  src = fetchgit {
+  src = fetchFromGitHub {
+    owner = "sclevine";
+    repo = "yj";
     inherit rev;
-    url = "https://github.com/sclevine/yj";
-    sha256 = "04irphzs6hp9hvyski29ad29ga1kis3h8bw7jqvmy2c2rkrrsh7x";
+    sha256 = "0bnb88wfm2vagh4yb1h9xhp3045ga0b6a77n3j2z5b4mvwshx5dr";
   };
 
-  goDeps = ./deps.nix;
+  vendorSha256 = "0y0n9fsb85qlpf9slwsxzarmfi98asa4x04qp2r8pagl28l0i8wv";
+
+  buildFlagsArray = [ "-ldflags=-X main.Version=${version}" ];
 
   meta = with stdenv.lib; {
     description = ''Convert YAML <=> TOML <=> JSON <=> HCL'';
```

Comment:
Why inherit version here? Did not see that before.

Reply:
When one calls `yj -v` to see the version of yj installed, `yj` prints the main.Version variable, that defaults to '0.0.0', so the real number is actually supplied at compile time. It also appears to be the way chosen by the developer of the package: https://github.com/sclevine/yj/search?q=Version

Reply:
``buildFlagsArray`` is fine. I mean the now removed line 25.